test/integ/quote-classic.test.ts

```diff
@@ -811,61 +860,297 @@ describe('quote', function () {
+                      if (!tokenOut.isNative) {
+                        expect(tokenOutAfter.subtract(tokenOutBefore).toExact()).to.equal(originalAmount);
+                      }
+
+                      // if the token out is native token, the difference will be slightly larger due to gas. We have no way to know precise gas costs in terms of GWEI * gas units.
+                      if (!tokenIn.isNative) {
+                        checkQuoteToken(
+                          tokenInBefore,
+                          tokenInAfter,
+                          CurrencyAmount.fromRawAmount(tokenIn, quoteJSON.quote)
+                        );
+                      }
+
+                      if (sendPortionEnabled) {
+                        const expectedPortionAmount = CurrencyAmount.fromRawAmount(
+                          tokenOut,
+                          quoteJSON.portionAmount ?? '0'
+                        );
+                        checkPortionRecipientToken(
+                          tokenOutPortionRecipientBefore!,
+                          tokenOutPortionRecipientAfter!,
+                          expectedPortionAmount
+                        );
+                      }
+                    }
+                  });
             });
           });
 
           GREENLIST_CARVEOUT_PAIRS.forEach(([tokenIn, tokenOut]) => {
             sendPortionEnabledValues.forEach((sendPortionEnabled) => {
-              it(`stable-to-stable ${tokenIn.symbol} -> ${tokenOut.symbol} carveout sendPortionEnabled = ${sendPortionEnabled}`, async () => {
-                // Arrange:
-                // - token amount from swapper
-                // - greenlist token pairs
-                // - parametrize on sendPortionEnabled
-                getAmount(1, type, tokenIn.symbol!, tokenOut.symbol!, '100');
-
-                // Act:
-                // - call classic quote
-
-                // Assert:
-                // - check if the response is 200
-                // - check if the response contains the methodParameters
-                // - check if the response contains only portionBips and portionAmount all equal to zero if sendPortionEnabled not send, explicitly check for portionRecipient = undefined
-
-                // Act:
-                // use the methodParameters to execute the swap
-
-                // Assert:
-                // - check if the swap is successful
-                // - check if the token balances are correct
-                //   - token balances assertion will stay the same for exact in and exact out
-                //   - need to explicitly check if the portion recipient balances has no change
-              });
+              // portion service doesn't have the stable-to-stable carveout merged yet
+              // all carveout tests are skipped for now
+              const shouldSkip = sendPortionEnabled;
```

Comment:
this test seems to be very very similar to the one above. I was thinking, shall we maybe have them all under one test, and make sure we can test right before the test is fired or not if its carveout or not, and based off that return the appropriate things?

Reply:
agreed; also, what we maybe should do is just have one test for all tokens and detect which ones are GREENList or not, and then expect the same results as if they did not have portion enabled.

It would be something like 

`if !portionEnabled || (portionEnabled && carveOut)`

Something like this